cachito/workers/scm.py

```diff
@@ -147,7 +167,23 @@ def fetch_source(self):
             ) as exc:
                 log.warning("Error handling archived artifact '%s': %s", previous_archive, exc)
 
-        self.clone_and_archive()
+        self.clone_and_archive(gitsubmodule)
+
+    def update_git_submodule(self, repo):
+        """Update git submodules.
+
+        For the given repo, update submodules. For Cachito request for `retrodep` repo and
+        `go-github` submodule, the dir structure would look like,
+        retrodep/go-github/<content_of_go-github_repo>
+
+        :param git.Repo repo: the repository object.
+        """
+        try:
+            log.debug(f"Git submodules for the requested repo are: {repo.submodules}")
+            repo.submodule_update()
```

Comment:
If I'm not mistaken, this function ends up calling [RootModule.update()](https://gitpython.readthedocs.io/en/stable/reference.html#git.objects.submodule.root.RootModule.update), which is recursive by default (fetches submodules inside submodules). Do we want that or not?

Reply:
AFAIR, We decided to not support nested gitsubmodules as of now. Adding a flag like that might make it more complicated when it comes to dependency handling. I'd refrain from opting that for now. 

Regarding the recursive thing, I did find that in the documentation initially but had thought of leaving it as it is since it's not adversely affecting something. but if there is a good reason to set it to `recursive=False`, I don't mind doing that either (if it is supported here)

Reply:
I'm not sure actually, would it be a problem if we cloned all submodules recursively but only included the top-level ones in the content manifest? My gut feeling says we should only clone what we declare in the content manifest, but IDK

Reply:
That seems fair. Let me test with setting `recursive=False`

Reply:
I updated it, however FWIW, `recursive=False` didn't make a difference. The submodules within submodules were not cloned anyhow. One could test it with,
`curl -X POST -H "Content-Type: application/json" http://localhost:8080/api/v1/requests -d '{"repo": "https://github.com/nirzari/retrodep.git", "ref": "d2a9933ca704a5b2120cd493af08d2d3aaa47ec2", "pkg_managers": ["gomod", "git-submodule"]}'`

Reply:
It looks like you did not update the "tour" submodule to the latest commit in your repo, you can do that using `git submodule update --remote -- tour` (you may need to run `git submodule init` first).

When I updated the submodule to the latest commit, the recursive option behaved as expected.